modules/swagger-codegen/src/main/java/io/swagger/codegen/languages/NodeJSGoogleCloudFunctionsCodegen.java

```diff
@@ -0,0 +1,92 @@
+package io.swagger.codegen.languages;
+
+import io.swagger.codegen.CliOption;
+import io.swagger.codegen.CodegenConfig;
+import io.swagger.codegen.SupportingFile;
+import io.swagger.models.Swagger;
+
+import java.util.Arrays;
+
+public class NodeJSGoogleCloudFunctionsCodegen extends AbstractNodeJSCodegen implements CodegenConfig {
+
+    public static final String EXPORTED_NAME = "exportedName";
+
+    public NodeJSGoogleCloudFunctionsCodegen() {
+        super();
+
+        // set the output folder here
+        outputFolder = "generated-code/nodejs-cloud-functions";
+
+        /*
+         * Template Location.  This is the location which templates will be read from.  The generator
+         * will use the resource stream to attempt to read the templates.
+         */
+        embeddedTemplateDir = templateDir = "nodejs-google-cloud-functions";
+
+        /*
+         * Supporting Files.  You can write single files for the generator with the
+         * entire object tree available.  If the input file has a suffix of `.mustache
+         * it will be processed by the template engine.  Otherwise, it will be copied
+         */
+        // supportingFiles.add(new SupportingFile("controller.mustache",
+        //   "controllers",
+        //   "controller.js")
+        // );
+        supportingFiles.add(new SupportingFile("swagger.mustache",
+                        "api",
+                        "swagger.yaml")
+        );
+        writeOptional(outputFolder, new SupportingFile("index.mustache", "", "index.js"));
+        writeOptional(outputFolder, new SupportingFile("package.mustache", "", "package.json"));
+        writeOptional(outputFolder, new SupportingFile("README.mustache", "", "README.md"));
+        if (System.getProperty("noservice") == null) {
```

Comment:
@jmuk thanks for the PR. Instead of using `noservice` system property, what about using `java -Dmodels -DsupportingFiles` instead? https://github.com/swagger-api/swagger-codegen#selective-generation

You may also consider using `.swagger-codegen-ignore` to skip certain files when regenerating the code: https://github.com/swagger-api/swagger-codegen#ignore-file-format